contracts/rewards/RewardsController.sol

```diff
@@ -41,12 +41,16 @@ contract RewardsController is RewardsDistributor, VersionedInitializable, IRewar
     _;
   }
 
+  address public ADDRESSES_PROVIDER;
+
+  constructor(address emissionManager) RewardsDistributor(emissionManager) {}
+
   /**
    * @dev Initialize for RewardsController
-   * @param emissionManager address of the EmissionManager
+   * @param provider The address of the PoolAddressesProvider
    **/
-  function initialize(address emissionManager) external initializer {
-    _setEmissionManager(emissionManager);
+  function initialize(address provider) external initializer {
+    ADDRESSES_PROVIDER = provider;
```

Comment:
This is a good point.
If the Rewards contracts can be used by multiple Aave Pools at the same time, then: i) having the same emissionManager shouldn't be problematic ii) linking the rewards contract to a specific Pool's PoolAddressesProvider contract does not make so much sense to me (would be better if we use a separate ProxyAdmin contract).

An alternative is keeping the same implementation of the Rewards contract and deploy a proxy per Aave Pool. This approach will give us more flexibility in terms of emissionManager roles, but at the cost of having 2 more contracts per pool (rewards and emissionManager)

Reply:
If emissionManager & rewardsController are always 1to1, why have 2 contracts in first place?

Reply:
Yes. The EmissionManager contract allows to allowlist an entity to govern the rewards for a particular asset (aToken/debtToken).

Its not an issue of having the same emissionManager for multiple pools at the same time because entities are allowlisted per asset (aToken/debtToken) and the address of these assets are different from one pool to another.